src/FixedDataTableWidthHelper.js

```diff
@@ -13,142 +13,93 @@
-    );
-    var newColumnWidth = Math.floor(column.props.width + columnFlexWidth);
+
+    const columnFlexWidth = Math.floor(
+      column.flexGrow / remainingFlexGrow * remainingFlexWidth);
+    const newColumnWidth = Math.floor(column.width + columnFlexWidth);
     totalWidth += newColumnWidth;
 
-    remainingFlexGrow -= column.props.flexGrow;
+    remainingFlexGrow -= column.flexGrow;
     remainingFlexWidth -= columnFlexWidth;
 
-    newColumns.push(React.cloneElement(
-      column,
-      {width: newColumnWidth}
-    ));
-  }
+    columnWidths.push(newColumnWidth);
+  });
 
   return {
-    columns: newColumns,
+    columnWidths,
     width: totalWidth,
   };
 }
 
-function adjustColumnGroupWidths(
-  /*array*/ columnGroups,
-  /*number*/ expectedWidth
-) /*object*/ {
-  var allColumns = [];
-  var i;
-  for (i = 0; i < columnGroups.length; ++i) {
-    React.Children.forEach(
-      columnGroups[i].props.children,
-      (column) => {
-        allColumns.push(column);
-      }
-    );
+function adjustColumnGroupWidths(columnGroups, expectedWidth) {
+  const allColumns = [];
+  forEach(columnGroups, columnGroup => {
+    Array.prototype.push.apply(allColumns, columnGroup.columns)
+  });
+
+  var remainingFlexGrow = getTotalFlexGrow(allColumns);
+  if (remainingFlexGrow === 0) {
+    return allColumns;
   }
+
   var columnsWidth = getTotalWidth(allColumns);
-  var remainingFlexGrow = getTotalFlexGrow(allColumns);
   var remainingFlexWidth = Math.max(expectedWidth - columnsWidth, 0);
-
-  var newAllColumns = [];
-  var newColumnGroups = [];
-
-  for (i = 0; i < columnGroups.length; ++i) {
-    var columnGroup = columnGroups[i];
-    var currentColumns = [];
-
-    React.Children.forEach(
-      columnGroup.props.children,
-      (column) => {
-        currentColumns.push(column);
-      }
-    );
-
-    var columnGroupFlexGrow = getTotalFlexGrow(currentColumns);
-    var columnGroupFlexWidth = Math.floor(
-      columnGroupFlexGrow / remainingFlexGrow * remainingFlexWidth
-    );
+  forEach(columnGroups, columnGroup => {
+    const currentColumns = columnGroup.columns;
+    const columnGroupFlexGrow = getTotalFlexGrow(currentColumns);
+    const columnGroupFlexWidth = Math.floor(
+      columnGroupFlexGrow / remainingFlexGrow * remainingFlexWidth);
 
     var newColumnSettings = distributeFlexWidth(
-      currentColumns,
-      columnGroupFlexWidth
-    );
-
+      currentColumns, columnGroupFlexWidth, columnGroupFlexGrow);
     remainingFlexGrow -= columnGroupFlexGrow;
     remainingFlexWidth -= columnGroupFlexWidth;
 
-    for (var j = 0; j < newColumnSettings.columns.length; ++j) {
-      newAllColumns.push(newColumnSettings.columns[j]);
-    }
-
-    newColumnGroups.push(React.cloneElement(
-      columnGroup,
-      {width: newColumnSettings.width}
-    ));
-  }
-
-  return {
-    columns: newAllColumns,
-    columnGroups: newColumnGroups,
-  };
-}
-
-function adjustColumnWidths(
```

Comment:
I've removed the case for handling no column groups.  Instead I use a default column group if none exists.